source/crud/tests/unified/client-bulkWrite-partialResults.yml

```diff
@@ -0,0 +1,238 @@
+description: "client bulkWrite partial results"
+schemaVersion: "1.4" # To support `serverless: forbid`
+runOnRequirements:
+  - minServerVersion: "8.0"
+    serverless: forbid
+
+createEntities:
+  - client:
+      id: &client0 client0
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database0Name crud-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collection0Name coll0
+
+initialData:
+  - collectionName: *collection0Name
+    databaseName: *database0Name
+    documents:
+      - &existingDocument { _id: 1, x: 11 }
+
+_yamlAnchors:
+  namespace: &namespace "crud-tests.coll0"
+  newDocument: &newDocument { _id: 2, x: 22 }
+
+tests:
+  - description: "partialResult is unset when first operation fails during an ordered bulk write (verbose)"
+    operations:
+      - object: *client0
+        name: clientBulkWrite
+        arguments:
+          models:
+            - insertOne:
+                namespace: *namespace
+                document: *existingDocument
+            - insertOne:
+                namespace: *namespace
+                document: *newDocument
+          ordered: true
+          verboseResults: true
+        expectError:
+          expectResult:
+            $$unsetOrMatches: {}
```

Comment:
> Test passes in the C driver unified test runner without implementation changes.

The same is true for the Java driver. The changes proposed by @kevinAlbs is very much needed.

Reply:
The Rust driver does not treat this document as a root-level document, so the test failed as expected before code changes. The UTR spec does not list this field in its (non-exhaustive) [list](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.md#allowing-extra-fields-in-root-level-documents) of root-level documents, and this line makes me think that we should _not_ be treating `expectedError.expectResult` as root-level, given that tests should likely always be asserting on the entire result object:

>The intention behind this rule is that it is not always feasible or relevant for a test to exhaustively specify all fields in an expected document (e.g. cluster time in command for [CommandStartedEvent](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.md#expectedEvent_commandStartedEvent)).

Do Java/C treat any top-level document as a root-level document? I expect there may be discrepancies between drivers in how this is handled.

Reply:
I expect other drivers may have similar behavior to C and Java. [$$unsetOrMatches](https://github.com/mongodb/specifications/blob/64c1692b8872301e552f04e6532029ac65666861/source/unified-test-format/unified-test-format.md#unsetormatches) does not seem to say if the match should be considered root or non-root. I initially expected the behavior to depend on the placement. E.g. `{ $$unsetOrMatches: {}}` to match as root, and `{ nested: { $$unsetOrMatches: {}}` to match as non-root.

A new `$$unset` operator might be a better fit to ensure `partialResults` is unset. However, to limit scope of this PR, I still recommend adding `$$exists: false` to prevent false positives.

Reply:
> However, to limit scope of this PR, I still recommend adding $$exists: false to prevent false positives.

Agreed, seems like there's some ambiguity here. Updated, the new test passes in Rust.